api/pkg/artifact/artifact.go

```diff
@@ -9,25 +9,33 @@ import (
 )
 
 type gcsClient struct {
-	API    *storage.Client
-	Config Config
+	API *storage.Client
 }
+
+type nopArtifact struct{}
+
 type Config struct {
 	Ctx context.Context
 }
 
 type Service interface {
-	DeleteArtifact(url string) error
+	DeleteArtifact(ctx context.Context, url string) error
 }
 
-func NewGcsClient(api *storage.Client, cfg Config) Service {
+func NewGcsArtifactService(api *storage.Client) Service {
 	return &gcsClient{
-		API:    api,
-		Config: cfg,
+		API: api,
 	}
 }
 
-func (gc *gcsClient) DeleteArtifact(url string) error {
+func NewNopArtifactService() Service {
+	return &nopArtifact{}
+}
+
+func (ad *nopArtifact) DeleteArtifact(ctx context.Context, url string) error {
+	return nil
+}
+func (gc *gcsClient) DeleteArtifact(ctx context.Context, url string) error {
```

Comment:
Minor nit: can we group all the implementations of the `artifact.Service` interface together? i.e.

```
interface definition

implementation 1 definition

implementation 2 definition
```

That way it's easier to look through the file for the various implementations 😁